CONTRIBUTING.rst

```diff
@@ -76,7 +76,37 @@ Before you submit a pull request, check that it meets these guidelines:
 
 1. The pull request should include tests.
 2. If the pull request adds functionality, the docs should be updated. Put your new functionality into a function with a docstring.
-3. If checks to not pass, have a look at https://travis-ci.com/pyfar/pyfar/pull_requests for more information.
+3. If checks do not pass, have a look at https://app.circleci.com/pipelines/github/pyfar/pyfar for more information.
+
+Function and Class Guidelines
+-----------------------------
+
+Functions and classes should
+
+* have a single clear purpose and a functionality limited to that purpose. Conditional arguments are fine in some cases but are an indicator that a function or class does not have a clear purpose. Conditional arguments are
+
+  - arguments that are obsolete if another argument is provided
+  - arguments that or necessary only if another argument is provided
+  - arguments that must have a specific value depending on other arguments
+
+* be split into multiple functions or classes if their purpose is not clear or the functionality not well limited.
```

Comment:
```suggestion

```
if the purpose is not clear we should maybe remove it.

Reply:
Good point

Reply:
True, I removed 'if their purppose is not clear'